terminology/terminology_processor.bal

```diff
@@ -0,0 +1,1661 @@
+                diagnostic = string `Allowed maximum size of output is: ${TERMINOLOGY_SEARCH_MAXIMUM_COUNT}; therefore, reduce the value of size parameter accordingly`,
+                errorType = r4:PROCESSING_ERROR,
+                httpStatusCode = http:STATUS_PAYLOAD_TOO_LARGE
+                );
+            }
+
+            // Validate whether the requested search parameters in the allowed list
+            foreach var param in searchParameters.clone().keys() {
+                if !CODESYSTEMS_SEARCH_PARAMS.hasKey(param) {
+                    return r4:createFHIRError(
+                string `This search parameter is not implemented yet: ${param}`,
+                r4:ERROR,
+                r4:PROCESSING_NOT_SUPPORTED,
+                diagnostic = string `Allowed search parameters: ${CODESYSTEMS_SEARCH_PARAMS.keys().toBalString()}`,
+                errorType = r4:VALIDATION_ERROR);
+                }
+            }
+
+            if self.finder is Finder {
+                return check (<Finder>self.finder).searchCodeSystem(searchParameters.clone(), offset = offset, count = count);
+            } else {
+                i4:CodeSystem[] codeSystemArray = self.codeSystems.toArray();
+
+                foreach var searchParam in searchParameters.clone().keys() {
+                    r4:RequestSearchParameter[] searchParamValues = searchParameters.clone()[searchParam] ?: [];
+
+                    i4:CodeSystem[] filteredList = [];
+                    if searchParamValues.length() != 0 {
+                        foreach var queriedValue in searchParamValues {
+                            i4:CodeSystem[] result = from i4:CodeSystem entry in codeSystemArray
+                                where entry[CODESYSTEMS_SEARCH_PARAMS.get(searchParam)] == queriedValue.value
+                                select entry;
+                            filteredList.push(...result);
+                        }
+                        codeSystemArray = filteredList;
+                    }
+                }
+
+                int total = codeSystemArray.length();
+
+                if total > TERMINOLOGY_SEARCH_MAXIMUM_COUNT {
+                    return r4:createFHIRError(
+                        "The response size is too large",
+                        r4:ERROR,
+                        r4:PROCESSING_TOO_COSTLY,
+                        diagnostic = "The response size is too large; therefore search with more specific parameter values",
+                        errorType = r4:PROCESSING_ERROR,
+                        httpStatusCode = http:STATUS_INTERNAL_SERVER_ERROR
+                        );
+                } else if total >= offset + count {
+                    return codeSystemArray.slice(offset, offset + count).clone();
+                } else if total >= offset {
+                    return codeSystemArray.slice(offset).clone();
+                } else {
+                    return [];
+                }
```

Comment:
Can't we consider the default behavior to provide TERMINOLOGY_SEARCH_MAXIMUM_COUNT number of results and handle all the cases with offset and count ?

Reply:
Tracked via: https://github.com/wso2-enterprise/open-healthcare/issues/1367